File: Difficulty: Basic/Array Subset/array-subset.java
//{ Driver Code Starts
// Initial Template for Java

// Initial Template for Java

import java.io.*;
import java.lang.*;
import java.util.*;

class GFG {
    public static void main(String[] args) throws IOException {
        BufferedReader read = new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(read.readLine().trim());
        while (t-- > 0) {
            String line = read.readLine().trim();
            String[] numsStr = line.split(" ");
            int[] a = new int[numsStr.length];
            for (int i = 0; i < numsStr.length; i++) {
                a[i] = Integer.parseInt(numsStr[i]);
            }
            line = read.readLine().trim();
            numsStr = line.split(" ");
            int[] b = new int[numsStr.length];
            for (int i = 0; i < numsStr.length; i++) {
                b[i] = Integer.parseInt(numsStr[i]);
            }
            Solution ob = new Solution();
            if (ob.isSubset(a, b))
                System.out.println("true");
            else
                System.out.println("false");
            System.out.println("~");
        }
    }
}

// } Driver Code Ends



class Solution {
    public boolean isSubset(int a[], int b[]) {
        // Your code here
        Set<Integer> setA = new HashSet<>();
        
        for(int num : a){
            setA.add(num);
        }
        
        Set<Integer> setB = new HashSet<>();
        
        for(int num : b){
            setB.add(num);
        }
        
        for(int num : setB){
            if(!setA.contains(num)){
                return false;
            }
        }
        return true;
    }
}

File: Difficulty: Basic/Array Subset/array-subset.py
#User function Template for python3

class Solution:
    #Function to check if a is a subset of b.
    def isSubset(self, a, b):
        # Your code here
        n = len(a)
        m = len(b)
        
        freq = {}
        
        for num in a:
            if num in freq:
                freq[num] += 1
            else:
                freq[num] = 1
                
        is_subset = True
        
        for num in b:
            if num in freq and freq[num] > 0:
                freq[num] -= 1
            else:
                is_subset = False
                
        
        return is_subset
    
    
    
    
